tests/test_conda_lock.py

```diff
@@ -177,3 +184,164 @@ def test_aggregate_lock_specs():
             platform="linux-64",
         ).env_hash()
     )
+
+
+def _create_conda_env(conda: PathLike, name: str) -> bool:
+    args: MutableSequence[PathLike] = [
+        str(conda),
+        "create",
+        "--name",
+        name,
+    ]
+
+    proc = subprocess.run(
+        args,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        encoding="utf8",
+    )
+
+    def print_proc(proc):
+        print(f"    Command: {proc.args}")
+        if proc.stdout:
+            print(f"    STDOUT:\n{proc.stdout}")
+        if proc.stderr:
+            print(f"    STDERR:\n{proc.stderr}")
+
+    try:
+        proc.check_returncode()
+    except subprocess.CalledProcessError:
+        try:
+            err_json = json.loads(proc.stdout)
+            message = err_json["message"]
+        except json.JSONDecodeError as e:
+            print(f"Failed to parse json, {e}")
+            message = ""
+
+        print("Could not perform conda create")
+        if message:
+            print(message)
+        print_proc(proc)
+
+        return False
+    return True
+
+
+def _destroy_conda_env(conda: PathLike, name: str) -> bool:
+    args: MutableSequence[PathLike] = [str(conda), "remove", "--name", name, "--all"]
+
+    proc = subprocess.run(
+        args,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        encoding="utf8",
+    )
+
+    def print_proc(proc):
+        print(f"    Command: {proc.args}")
+        if proc.stdout:
+            print(f"    STDOUT:\n{proc.stdout}")
+        if proc.stderr:
+            print(f"    STDERR:\n{proc.stderr}")
+
+    try:
+        proc.check_returncode()
+    except subprocess.CalledProcessError:
+        try:
+            err_json = json.loads(proc.stdout)
+            message = err_json["message"]
+        except json.JSONDecodeError as e:
+            print(f"Failed to parse json, {e}")
+            message = ""
+
+        print("Could not perform conda remove")
+        if message:
+            print(message)
+        print_proc(proc)
+
+        return False
+    return True
+
+
+@pytest.fixture
+def conda_exe():
+    return determine_conda_executable("conda", no_mamba=True)
+
+
+@pytest.fixture
+def conda_env(conda_exe):
```

Comment:
I don't think we need this fixture, since this tests that we can install things in a way we're not using?

Reply:
we can use the destroy with a finally in the test